src/common.h

```diff
@@ -369,6 +369,24 @@ char *ly_strnchr(const char *s, int c, size_t len);
  */
 int ly_strncmp(const char *refstr, const char *str, size_t str_len);
 
+/**
+ * @brief Get xml namespace prefix or just some substring from @p str.
+ *
+ * The function looking for possible xml namespace prefix at the beginning of the @p str_begin and return it.
+ * If there is some string at the @p str_begin and then follows the prefix, then the string is returned.
+ * If @p str_begin does not contain any prefix, then the entire @p str_begin is returned.
+ * The @p rest can be used as the @p str_begin in the next iteration.
+ *
+ * @param[in] str_begin Begin of the input string.
+ * @param[in] str_end End of the input string. If set to NULL then the @p str_begin must be NULL-terminated string.
+ * @param[out] substr_type Type of substring found. Set to 1 for the xml namespace prefix, otherwise 0.
```

Comment:
Either call it `is_prefix`, for example, or change the type to an enum. Preferably the former.